app/CMakeLists.txt

```diff
@@ -196,23 +205,36 @@ if(CELERITAS_BUILD_DEMOS AND CELERITAS_USE_VecGeom)
       demo-loop/LDemoKernel.cu
     )
   endif()
-  add_library(celeritas_demo_loop
+  celeritas_add_library(celeritas_demo_loop
     demo-loop/LDemoIO.cc
     demo-loop/LDemoParams.cc
     demo-loop/LDemoRun.cc
     ${_cuda_src}
   )
 
-  target_link_libraries(celeritas_demo_loop
+  target_link_libraries(celeritas_demo_loop_final
     PRIVATE
-    celeritas
+    Celeritas::Core
+    nlohmann_json::nlohmann_json
+    Celeritas::ROOT
+  )
+  target_link_options(celeritas_demo_loop_final
+     PRIVATE $<DEVICE_LINK:$<TARGET_FILE:celeritas_static>>
+  )
+  target_link_libraries(celeritas_demo_loop_objects
+    PRIVATE
+    Celeritas::Core
     nlohmann_json::nlohmann_json
     Celeritas::ROOT
   )
   if(CELERITAS_USE_CUDA)
     celeritas_link_vecgeom_cuda(celeritas_demo_loop)
   endif()
 
+  if(CELERITAS_USE_CUDA)
+    celeritas_link_vecgeom_cuda(celeritas_demo_loop_final)
+  endif()
```

Comment:
Roll this into the `if` statement on 231 -- and are both really needed?

Reply:
Agreed, however I am hoping to get rid of the repetition.